pymongo/helpers.py

```diff
@@ -221,7 +221,9 @@ def _check_write_command_response(result):
     if write_errors:
         _raise_last_write_error(write_errors)
 
-    error = result.get("writeConcernError")
+    error = result.get("writeConcernError", {})
+    error_labels = result.get("errorLabels", [])
+    error.update({'errorLabels': error_labels})
```

Comment:
We should ensure that `_check_command_response` above has the same behavior. Also is it ever possible for `error` to be  RawBSONDocument (where `.update` would fail)?

Reply:
Added the behavior to `_check_command_response`. The question about `RawBSONDocument` needs further investigation.

Reply:
This logic is flawed because it ends up always making `error` true. It should be:
```python
    error = result.get("writeConcernError", {})
    if error:
        error_labels = result.get("errorLabels", [])
        if error_labels:
            error.update({'errorLabels': error_labels})
        _raise_write_concern_error(error)
```

Reply:
Good catch. Thanks.

Reply:
This is not resolved because we still need to answer the RawBSONDocument question.